src/components/NanoContract/NanoContractTransactionsListHeader.component.js

```diff
@@ -0,0 +1,265 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useState } from 'react';
+import {
+  StyleSheet,
+  View,
+  Text,
+  TouchableWithoutFeedback,
+  Linking,
+} from 'react-native';
+import { useSelector } from 'react-redux';
+import { t } from 'ttag';
+import { COLORS } from '../../styles/themes';
+import { combineURLs, getShortContent, getShortHash } from '../../utils';
+import SimpleButton from '../SimpleButton';
+import HathorHeader from '../HathorHeader';
+import { NanoContractIcon } from '../Icons/NanoContract.icon';
+import { ArrowDownIcon } from '../Icons/ArrowDown.icon';
+import { ArrowUpIcon } from '../Icons/ArrowUp.icon';
+import { TextValue } from '../TextValue';
+import { TextLabel } from '../TextLabel';
+import { EditInfoContainer } from '../EditInfoContainer';
+import { SelectAddressModal } from './SelectAddressModal.component';
+import { UnregisterNanoContractModal } from './UnregisterNanoContractModal.component';
+
+/**
+ * It presents the header for the list of transactions of a Nano Contract screen.
+ *
+ * @param {Object} props
+ * @param {Object} props.nc Nano Contract data
+ * @param {string} props.address Default address selected
+ * @param {(address:string) => {}} props.onAddressChange Function called when address changes
+ */
+export const NanoContractTransactionsListHeader = ({ nc, address, onAddressChange }) => {
+  const [isShrank, toggleShrank] = useState(true);
+  const [selectedAddress, setSelectedAddress] = useState(address);
+  const [showSelectAddressModal, setShowSelectAddressModal] = useState(false);
+  const [showUnregisterNanoContractModal, setShowUnregisterNanoContractModal] = useState(false);
+
+  const isExpanded = () => !isShrank;
+
+  const onEditAddress = () => {
+    setShowSelectAddressModal(true);
+  };
+
+  const toggleSelectAddressModal = () => {
+    setShowSelectAddressModal(!showSelectAddressModal);
+  };
+
+  const onUnregisterNanoContract = () => {
+    setShowUnregisterNanoContractModal(true);
+  };
+
+  const toggleUnregisterNanoContractModal = () => {
+    setShowUnregisterNanoContractModal(!showUnregisterNanoContractModal);
+  };
+
+  const hookSelectAddress = (pickedAddres) => {
```

Comment:
Misleading variable name, maybe rename it to `handleSelectAddress`

Reply:
Indeed! Changed.